dhis-2/dhis-services/dhis-service-dxf2/src/test/java/org/hisp/dhis/dxf2/metadata/objectbundle/ObjectBundleServiceAttributesTest.java

```diff
@@ -238,8 +238,7 @@ public void testValidateMetadataAttributeValuesUnique()
         ObjectBundleValidationReport validationReport = objectBundleValidationService.validate( bundle );
         List<ObjectReport> objectReports = validationReport.getObjectReports( DataElement.class );
 
-        assertFalse( objectReports.isEmpty() );
-        assertEquals( 2, validationReport.getErrorReportsByCode( DataElement.class, ErrorCode.E4009 ).size() );
```

Comment:
After debugging this for a long time I think the expectations here are actually wrong. This is what we would **like** to happen from the correctness and data consistency point of view but reading the code I don't see why this would happen. 
The scenario here is : the imported file contains 4 data elements. All refer to two unique attributes, one in the payload, one created in the test setup. The test setup also creates 3 data elements which get the unique attributes using the values: `Value1`, `Value2`, `Value3`. The 4 elements of the payload also use `Value1`, `Value2` so this is a violation of the unique but none of the 3 elements created in the test setup are part of the preheat so they are not visible when performing the unique check so I'd expect the outcome to be that with regards to the payload there is no unique violation because the violating elements are not visible in the preheat scope.